pkg/kubelet/container/helpers.go

```diff
@@ -318,7 +318,7 @@ func MakePortMappings(container *v1.Container) (ports []PortMapping) {
 		// this is necessary for rkt.
 		// http://issue.k8s.io/7710
 		if p.Name == "" {
-			pm.Name = fmt.Sprintf("%s-%s:%d", container.Name, p.Protocol, p.ContainerPort)
+			pm.Name = fmt.Sprintf("%s-%s%d:%d", container.Name, p.Protocol, p.HostPort, p.ContainerPort)
```

Comment:
I don't know the consequences of changing this string.  @Random-Liu or @yujuhong can you comment on this?  Specifically, if a pod is running using the old name and kubelet is restarted, is that a problem?

Reply:
If we're going to change this string, let's change it to `<name>-<proto>:<containerPort>:<hostPort>` ?

Reply:
Please also append ``<hostIP>``, while you're at it. (That doesn't solve the whole problem space discussed here either, but would be very useful on multihomed nodes.)

Reply:
@thockin It looks like this string was changed before without any additional changes and release notes: https://github.com/kubernetes/kubernetes/pull/82374. Not a definite indication that this is not breaking anything, just an observation at this point.

Reply:
ok, names are not being preserved and only used for dedup: https://github.com/kubernetes/kubernetes/pull/93581

So this should not be a concern. I am looking further into whether this dedup logic is even needed in kubelet or it can be offloaded to CNI.